src/app/contextBroker/contextBroker.cpp

```diff
@@ -1223,8 +1229,30 @@ const char* description =
 */
 static void contextBrokerInit(std::string dbPrefix, bool multitenant)
 {
+
+  Notifier                            *pNotifier = NULL;
+  SyncQOverflow<SenderThreadParams*>  *pQueue    = NULL;
+  QueueWorkers                        *qws       = NULL;
+  int                                 rc         = 0;
+
+  /* If we use a queue for notifications, start worker threads */
+  if (strcmp(notificationMode, "threadpool") == 0)
+  {
+    LM_T(LmtNotifier,("Setting up queue and threads for notifications"));
+    pQueue = new SyncQOverflow<SenderThreadParams*>(notificationQueueSize);
+    pNotifier = new QueueNotifier(pQueue);
+    qws = new QueueWorkers(pQueue, notificationThreadNum);
+    if ((rc = qws->start()) != 0){
+      LM_E(("Runtime Error starting notification queue workers (%d)", rc));
+    }
+  }
+  else
+  {
+    pNotifier = new Notifier();
+  }
+
```

Comment:
Why not make this code chunk a function and put it in the ngsiNotify library.
I imagine the reason SyncQOverflow is in common it that it is needed here.
If this is 'encapsulated' in ngsiNotify, perhaps all the 'notification-queue' code can go in ngsiNotify.

Actually, I wouldn't mind at all making a whole new library of the notification-queue,
doesn't have to be in this very PR, of course.

About style guide:
- The '*' for pointers, we put them with the *type*, not the *variable*.
- align '=' and the code gets esaier to read (here: pQueue, pNotifier and qws)
- '{' always on its own line, aligned with its '}'

And. lastly, if qws->start() fails, is there any reason to go on?
Would this mean the broker is running, unable to notify?
If so, it's probably better to just die. Something is terribly wrong ...